commcare_connect/opportunity/helpers.py

```diff
@@ -93,6 +87,62 @@ def started_learning():
         return Count("opportunityaccess", filter=Q(opportunityaccess__date_learn_started__isnull=False), distinct=True)
 
 
+def get_deliveries_count_subquery(status=None):
+    filters = {"opportunity_access__opportunity_id": OuterRef("pk")}
+    if status is not None:
+        filters["status"] = status
+
+    return Coalesce(
+        Subquery(
+            CompletedWork.objects.filter(**filters)
+            .values("opportunity_access__opportunity_id")
+            .annotate(count=Count("id", distinct=True))
+            .values("count"),
+            output_field=IntegerField(),
+        ),
+        0,
+    )
+
+
+def total_accrued_sq():
+    return Coalesce(
+        Subquery(
+            OpportunityAccess.objects.filter(opportunity_id=OuterRef("pk"))
+            .values("opportunity_id")
+            .annotate(total=Sum("payment_accrued"))
+            .values("total"),
+            output_field=IntegerField(),
+        ),
+        0,
+    )
+
+
+def total_paid_sq():
+    return Coalesce(
+        Subquery(
+            Payment.objects.filter(opportunity_access__opportunity_id=OuterRef("pk"))
+            .values("opportunity_access__opportunity_id")
+            .annotate(total=Sum("amount"))
+            .values("total"),
+            output_field=IntegerField(),
+        ),
+        0,
+    )
+
+
+def deliveries_from_yesterday_sq():
+    return Coalesce(
+        Subquery(
+            UserVisit.objects.filter(opportunity_id=OuterRef("pk"), visit_date__gte=now().date() - timedelta(1))
```

Comment:
nit: use kwargs to avoid ambiguity:
```suggestion
            UserVisit.objects.filter(opportunity_id=OuterRef("pk"), visit_date__gte=now().date() - timedelta(days=1))
```

Reply:
changed - [b7776a8](https://github.com/dimagi/commcare-connect/pull/621/commits/b7776a88e2e6e27c8763f54e5ce4bc70f7b0f0c0)